asmcomp/cmmgen.ml

```diff
@@ -878,18 +882,24 @@ let box_int_constant bi n =
     Pnativeint -> Uconst_nativeint n
   | Pint32 -> Uconst_int32 (Nativeint.to_int32 n)
   | Pint64 -> Uconst_int64 (Int64.of_nativeint n)
+  | Puint32 -> Uconst_uint32 (Uint32.of_int32 (Nativeint.to_int32 n))
+  | Puint64 -> Uconst_uint64 (Uint64.of_int64 (Int64.of_nativeint n))
```

Comment:
I found it a bit surprising at first that `Uint64` would not have a `of_nativeint` function if `Int64` has one (are the two modules not supposed to share a common interface on the non-sign-related parts?). I suppose that one of the reason is that the natural implementation is `fun n -> of_64 (Int64.of_nativeint n)`, and that you wanted to avoid having `UInt64` depend on `Int64`? In general, is there a rationale for which conversion functions should or should not be provided by each module? (Why would `Nativeint.to_uint32` not exist, for example?)